body_test.go

```diff
@@ -0,0 +1,172 @@
+package artifact
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+var allTheBytes = []byte{1, 3, 7, 15, 31, 63, 127, 255}
+
+func setup(t *testing.T) (*os.File, []byte, func()) {
+
+	var b bytes.Buffer
+
+	file, err := ioutil.TempFile(".", "body-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	out := io.MultiWriter(file, &b)
+
+	for i := 0; i < 256; i++ {
+		_, err := out.Write(allTheBytes)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
```

Comment:
I used this particular pattern because it made finding errors easier, since as long as I read a certain chunksize, the beginning of each chunk/part would have the same sequence.

I'll change this and double check that the tests still work